### DISCUSSION TITLE
peek and untracked

### DISCUSSION THREAD

**Discussion by hcschuetz** [preactjs/signals#629] [Q&A] [answered]
Hi,
reading the docs and looking at the code I got the impression that `mySignal.peek()` is equivalent to `untracked(() => mySignal.value)`.

Is this true?

If it is, it might be helpful to mention this fact in the README because it would help new users building their mental model of signals.  And if it isn't, it might also be helpful to explain whatever subtle differences there are.

And a related point:
I wanted to suggest this improvement not only for the README but also for the docs, but apparently `untracked` isn't documented at all.  Does this mean that `untracked` is an undocumented feature that should not be used?

Regards,
Heribert

**Comment by JoviDeCroock** [comment, marked as answer]
This is true, `untracked` is slightly cheaper with regards to how the tracking-context is shifted and can `.peek()` multiple values at once.

I have our documentation on my todo-list, basically for signals splitting it up into core - preact - react and more generally the Preact docs.

**Reply by hcschuetz** [reply]
Thanks for letting me know and for answering so quickly!

**Comment by hcschuetz** [comment]
Another thought:
One might say something like

> `mySignal.peek()` is a shorthand for `untracked(() => mySignal.value)` if you need untracked access only to a single signal.

And actually `.peek()` might even be implemented that way.
